alerts/cluster-monitoring-operator/TargetDown.md

```diff
@@ -0,0 +1,269 @@
+  jobLabel: app.kubernetes.io/name
+  namespaceSelector: {}
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: exporter
+      app.kubernetes.io/name: node-exporter
+      app.kubernetes.io/part-of: openshift-monitoring
+```
+In the `Prometheus` that scrapes this target, the volume mounted at `/etc/prometheus/secrets/*`
+are the secrets having the same name as the subdirectory. In this example, the secret
+is `metrics-client-certs` in the namespace `openshift-monitoring`. Now we extract
+the certificate from the secret using this command:
+```bash
+oc extract secret/$SECRET_NAME -n $NAMESPACE --keys=tls.crt --to=- > certificate.crt
+```
+Then we inspect is expiration date.
+```bash
+openssl x509 -noout -enddate -in certificate.crt
+```
+The output should contain `notAfter` field as its expiration date.
+```bash
+notAfter=Aug  6 13:11:20 2025 GMT
+```
+
+Normally Openshift can automatically renew the certificates before expiration date.
+However, if the certificate does expire, we shall force to renew it manually:
+1. Create a backup of the secret which contains the certificate.
+```bash
+oc get secret $SECRET_NAME -n $NAMESPACE -o yaml &> secret-tls.yaml
+```
+2. Delete the secret holding the certificate.
+```bash
+oc delete secret $SECRET_NAME -n $NAMESPACE
+```
+3. Check the newly created secret contains a valid certificate.
+```bash
+oc get secret $SECRET_NAME -n $NAMESPACE -o yaml
+```
+It should return certificate content with expiration date.
+```bash
+apiVersion: v1
+data:
+  tls.crt: ....
+  tls.key: ....
+kind: Secret
+metadata:
+  annotations:
+    service.alpha.openshift.io/expiry: "2025-08-06T13:11:20Z"     <<< Here is the expiration date.
+    service.beta.openshift.io/expiry: "2025-08-06T13:11:20Z"
+    service.beta.openshift.io/originating-service-name: prometheus-k8s
+    service.beta.openshift.io/originating-service-uid: a781e552-b3c3-4b74-b068-1d7152671b72
+  name: prometheus-k8s-tls
+  namespace: openshift-monitoring
+  ownerReferences:
+  - apiVersion: v1
+    kind: Service
+    name: prometheus-k8s
+type: kubernetes.io/tls
+```
+4. Restart the pods using this certificate by deleting the pods.
+```bash
+# by name
+oc delete pod $POD_NAME -n $NAMESPACE
+# or by label
+oc delete pod -l $LABEL_SELECTOR -n $NAMESPACE
+```
+
+### Prometheus Scraping Interval
+
+Ensure that the scraping interval for the target is appropriately configured.
+Too frequent scraping can degrade the performance of target pod, even causing
+timeout when scraping.
+
+Pay close attention to the metric `scrape_duration_seconds`. If scrape duration
+is close to the scrape interval, we may consider to increase the interval.
+
+### Target Metrics Path
+
+Verify that the metrics path in the target is correctly defined in `ServiceMonitor`
+and `PodMonitor`. If due to recent update the target changes the metrics path, the
+metric endpoint should update accordingly.
+
+### Target Restart or Update
```

Comment:
we can ask to check this before running `its localhost:port/metrics` locally in the container. (see above)